packages/amplify-category-function/src/commands/function/build.ts

```diff
@@ -31,7 +31,12 @@ export const run = async (context: $TSContext) => {
   } catch (err) {
     context.print.info(err.stack);
     context.print.error('There was an error building the function resources');
-    void context.usageData.emitError(err);
+    const amplifyError = new AmplifyError(
+      'PackagingLambdaFunctionError',
+      { message: `There was an error building the function resources ${err.message}` },
+      err,
+    );
+    void context.usageData.emitError(amplifyError);
     process.exitCode = 1;
```

Comment:
Should we just throw this instead of process.exitCode = 1 ?

Reply:
I was doing that but reverted after seeing that every top level command is doing the same. We need to tackle the `process.exitCode` in a separate PR and make sure it doesn't break anything